JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -121,103 +139,160 @@ fun HomeScreen(
-        val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
+                // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
+                // don't restart the effect and use the latest lambda values.
+                val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
+                val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
 
-        // Effect running in a coroutine that displays the Snackbar on the screen
-        // If there's a change to errorMessageText, retryMessageText or scaffoldState,
-        // the previous effect will be cancelled and a new one will start with the new values
-        LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
-            val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
-                message = errorMessageText,
-                actionLabel = retryMessageText
-            )
-            if (snackbarResult == SnackbarResult.ActionPerformed) {
-                onRefreshPostsState()
+                // Effect running in a coroutine that displays the Snackbar on the screen
+                // If there's a change to errorMessageText, retryMessageText or scaffoldState,
+                // the previous effect will be cancelled and a new one will start with the new values
+                LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
+                    val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
+                        message = errorMessageText,
+                        actionLabel = retryMessageText
+                    )
+                    if (snackbarResult == SnackbarResult.ActionPerformed) {
+                        onRefreshPostsState()
+                    }
+                    // Once the message is displayed and dismissed, notify the ViewModel
+                    onErrorDismissState(errorMessage.id)
+                }
+            }
+        } else {
+            val selectedPost = uiState.posts.find { it.id == uiState.selectedPostId }
+
+            if (selectedPost != null) {
+                val lazyListState = detailLazyListStates.getValue(selectedPost.id)
+
+                ArticleScreen(
+                    post = selectedPost,
+                    onBack = onInteractWithList,
+                    isFavorite = uiState.favorites.contains(selectedPost.id),
+                    onToggleFavorite = {
+                        onToggleFavorite(selectedPost.id)
+                    },
+                    lazyListState = lazyListState,
+                )
+
+                // If we are just showing the detail, have a back press switch to the list.
+                if (!useListDetail) {
+                    BackHandler {
+                        onInteractWithList()
+                    }
+                }
+            } else {
+                // TODO: Improve UX
+                LaunchedEffect(Unit) {
```

Comment:
Confusing, not sure what's happening here

Reply:
This is the same issue that the `ArticleScreen` currently has on its own: If after process death we get back to this screen and can't load the article, it'd be better to have an affordance to retry or go back rather than just jumping back to the list.